db/collection_index.go

```diff
@@ -179,8 +183,11 @@ func (c *collection) updateIndexedDoc(
 	if err != nil {
 		return err
 	}
+	// TODO-ACP: https://github.com/sourcenetwork/defradb/issues/2365 - ACP <> Indexing, possibly also check
+	// and handle the case of when oldDoc == nil (will be nil if inaccessible document).
```

Comment:
todo: Please add an error if trying to define an index on a collection with a policy.

Reply:
- [ ] Return an error when trying to use indexes with acp until #2365 

- [ ] Write a test to assert the error

Reply:
Discussed with Islam seems like disabling adding index is enough.

Disabled adding of index and test in this commit: ``` PR(ACP): Disable Creating Index ```